### ISSUE TITLE
request proxying/by-passing of timed-out go plugin, as if there is no plugin at all - leading to malicious/bad requests at upstream

### ISSUE THREAD

**Issue by diefans** [Kong/kong#8546]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

2.7.1

### Current Behavior

If an upstream is wrapped/guarded by a go plugin and the plugin takes longer than 60 seconds (btw. which timeout is that?), kong will just **"skip"** the plugin and proxy the request **"half-baked"** to the upstream.

So whatever the plugin was about to add or change on behalf of the request is missing or incomplete, leading to undefined behavior and breaking SLA/API between plugin and upstream.

Kong logs show:
```
kong_1                | 2022/03/16 14:41:29 [error] 1109#0: *719 [kong] pb_rpc.lua:365 [bm-authx] timeout, client: 192.168.33.1, server: kong, request: "GET /ucm/status HTTP/1.1", host: "example.com"
kong_1                | 192.168.33.1 - - [16/Mar/2022:14:41:29 +0000] "GET /ucm/status HTTP/1.1" 200 21 "-" "HTTPie/2.4.0"
```

### Expected Behavior

I would expect some sort of `504 Gateway Timeout`, but definitely nothing passed to upstream.

### Steps To Reproduce

```markdown
1. put some `time.Sleep(100 * time.Second)` somewhere in a go plugin
2. call a route affected by this sleep
```

### Anything else?

Kong times out the plugin in 60 seconds.

- Which setting is used to configure this timeout?
- Is there a way to stop by-passing the plugin and terminate the request with a `504`?

**Comment by diefans** [comment]
So the setting for the above timeout is https://github.com/openresty/lua-nginx-module#lua_socket_read_timeout
```
lua_socket_read_timeout 60s
```

**Comment by diefans** [comment]
So IMHO I think this function is "eating up" the timeout, where the rest of the stack has no clue about it: https://github.com/Kong/kong/blob/ce4e424377d7ce5c4cf51b33165174e18913bd1e/kong/runloop/plugin_servers/pb_rpc.lua#L357-L372

**Comment by Suika-Kong** [comment]
Thanks for reporting! It seems you find the reason, and maybe you could fix it?
PR is welcomed, and we have a T-shirt for new [contributors](https://github.com/Kong/kong/blob/master/CONTRIBUTING.md)!

**Comment by StarlightIbuki** [comment]
Seems fixed by #8531
